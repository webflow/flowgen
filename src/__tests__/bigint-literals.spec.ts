import { compiler, beautify } from "..";
import "../test-matchers";

it("should handle bigint literals in type", () => {
  const ts = `
  type MyBigType = bigint;
  type bigint_like = number | bigint | string;
  
  // these are not supported by flowgen yet
  // but we should test them once they are supported
  // const literal = 9007199254740991n;
  // const ctor = BigInt("9007199254740991"); // 9007199254740991n
`;

  const result = compiler.compileDefinitionString(ts, { quiet: true });

  expect(beautify(result)).toMatchSnapshot();
  expect(result).toBeValidFlowTypeDeclarations();
});
