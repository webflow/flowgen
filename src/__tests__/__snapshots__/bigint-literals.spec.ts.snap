// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle bigint literals in type 1`] = `
"declare type MyBigType = number;
declare type bigint_like = number | number | string;
"
`;
